File: kLargest.py
#Time complexity: O(nlogk)
#space complexity: O(k)
import heapq as hq
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        minH = []
        
        for num in nums:
            hq.heappush(minH, num)
            if len(minH) > k:
                hq.heappop(minH)
        return hq.heappop(minH)
        

File: kLargest.cpp
// Time Complexity :O(nlogk) n is the total elememnts
// Space Complexity : O(k) k-> Size of the heap  
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int,vector<int>,greater<int>> pq;
        for(auto num : nums){
            pq.push(num);
            if(pq.size() > k){
                pq.pop();
            }
        }
        
        return pq.top();
    }
};